### ISSUE TITLE
[Enhancement] Redesign Base Trainer for Strategy-Specific Builder Components

### ISSUE THREAD

**Issue by amina-mardiyyah** [ML4LitS/OTAR3088#97]
# Description
This task focuses on refactoring the current `HF base_trainer` to make it strategy-agnostic and reusable across different training approaches. 
Currently, the `base_trainer` script is tightly coupled with a single training flow. Since we're currently extending the pipeline to include other training strategies (e.g., Differential Learning Rate, Hyperparameter tuning, partial freezing etc),they would all share overlapping `hf` methods. Hence, there is significant code duplication. 

## Proposed Design

- Extract shared utilities into common base modules, functions

- Modularise the current `hf_trainer` func by introducing builder components for each training strategy.

- Each strategy  will extend this base trainer components via its own builder.

- The central `base_trainer.py` script would be agnostic to any strategy by: 

  - dynamically pulling the correct components using a defined **Model Registry** dict(which shows all supported strategies. 
  - Executes training with consistent logging and metrics tracking.

## To-Do
- [x] Create new scripts/modules to host common shared methods/functions
- [x] Refactor hf base_trainer_script
- [x] Strategy-specific builder functions.
- [x] Central script (`base_trainer.py`) that supports all strategies seamlessly.
- [x] `main.py` works as before, pulling the right stratgey via defined flag in config `training_strategy`
- [x] Update config file for new variables specific to each strategy and a flag for training strategy
- [x] Verified compatibility with existing logging logic
